proof_of_concept/definitions/workflows.py

```diff
@@ -236,16 +244,19 @@ def calc_step_outputs(
             step_hash = sha256()
             for inp_name in sorted(step.inputs):
                 inp_item = '{}.{}'.format(step.name, inp_name)
-                step_hash.update(item_keys[inp_item].encode('utf-8'))
-            step_hash.update(step.compute_asset_id.encode('utf-8'))
+                step_hash.update(
+                        item_keys[inp_item].encode('utf-8'))
+            step_hash.update(
+                    step.compute_asset_id.encode('utf-8'))
             for outp_name in step.outputs:
                 outp_item = '{}.{}'.format(step.name, outp_name)
                 outp_hash = step_hash.copy()
                 outp_hash.update(outp_name.encode('utf-8'))
-                item_keys[outp_item] = 'hash:{}'.format(outp_hash.hexdigest())
```

Comment:
Implementing the AssetId class revealed that `item_keys` had mostly keys in it, except for the step outputs, where it had asset ids. So I changed it to only have keys everywhere, and those keys get converted to AssetIds when needed (see the `from_key` calls above).

Reply:
Bueno!